pkg/statedb/reconciler/retries.go

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconciler
+
+import (
+	"bytes"
+	"container/heap"
+
+	"github.com/cilium/cilium/pkg/backoff"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+func newRetries(minDuration, maxDuration time.Duration) *retries {
+	const prealloc = 64
+	return &retries{
+		backoff: backoff.Exponential{
+			Min: minDuration,
+			Max: maxDuration,
+		},
+		queue: make([]*retryItem, 0, prealloc),
+		items: make(map[string]*retryItem, prealloc),
+	}
+}
+
+// retries holds the items that failed to be reconciled in
+// a priority queue ordered by retry time.
```

Comment:
maybe a note about this not being safe to concurrently access

Reply:
since this is internal I'm not sure that's needed.